File: sj-miaosha/src/main/java/com/nemo/sj/miaosha/common/cache/RedisConfig.java
package com.nemo.sj.miaosha.common.cache;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import org.springframework.stereotype.Component;

@Configuration
@ConfigurationProperties(prefix="redis")
public class RedisConfig {
	private String host;
	private int port;
	private int timeout=100;//秒
	private String password;
	private int poolMaxTotal=100;
	private int poolMaxIdle=500;
	private int poolMaxWait=20;//秒
	public String getHost() {
		return host;
	}
	public void setHost(String host) {
		this.host = host;
	}
	public int getPort() {
		return port;
	}
	public void setPort(int port) {
		this.port = port;
	}
	public int getTimeout() {
		return timeout;
	}
	public void setTimeout(int timeout) {
		this.timeout = timeout;
	}
	public String getPassword() {
		return password;
	}
	public void setPassword(String password) {
		this.password = password;
	}
	public int getPoolMaxTotal() {
		return poolMaxTotal;
	}
	public void setPoolMaxTotal(int poolMaxTotal) {
		this.poolMaxTotal = poolMaxTotal;
	}
	public int getPoolMaxIdle() {
		return poolMaxIdle;
	}
	public void setPoolMaxIdle(int poolMaxIdle) {
		this.poolMaxIdle = poolMaxIdle;
	}
	public int getPoolMaxWait() {
		return poolMaxWait;
	}
	public void setPoolMaxWait(int poolMaxWait) {
		this.poolMaxWait = poolMaxWait;
	}
}

File: sj-web/src/main/kotlin/com/nemo/sj/config/RedisConfig.kt
package com.nemo.sj.config

import com.nemo.sj.bean.KeyGeneratorImpl
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.redis.cache.RedisCacheManager
import org.springframework.cache.CacheManager
import org.springframework.data.redis.core.RedisTemplate



/**
 *  create by Nemo
 *  2018/6/9  10:20
 */
@Configuration
class RedisConfig {

    /**
     * 缓存管理器.
     * @param redisTemplate
     * @return
     */
//    @Bean
//    fun cacheManager(redisTemplate: RedisTemplate<*, *>): CacheManager {
//        return RedisCacheManager(redisTemplate)
//    }
    @Bean
    fun redisManage(): com.nemo.sj.cache.CacheManager{
        return com.nemo.sj.cache.CacheManager();
    }
}